var/spack/repos/builtin/packages/slate/package.py

```diff
@@ -25,34 +25,35 @@ class Slate(Package):
     variant('mpi',    default=True, description='Build with MPI support.')
     variant('openmp', default=True, description='Build with OpenMP support.')
 
+    depends_on('bash', type='build')
+    depends_on('scalapack')
+    depends_on('blas')
     depends_on('cuda@9:10', when='+cuda')
-    depends_on('intel-mkl')
-    depends_on('mpi', when='+mpi')
+    depends_on('mpi',       when='+mpi')
 
     conflicts('%gcc@:5')
 
-    def setup_build_environment(self, env):
-        if('+cuda' in self.spec):
-            env.prepend_path('CPATH', self.spec['cuda'].prefix.include)
-        env.prepend_path('CPATH', self.spec['intel-mkl'].prefix.mkl.include)
-
-    def install(self, spec, prefix):
-        f_cuda = "1" if spec.variants['cuda'].value else "0"
-        f_mpi = "1" if spec.variants['mpi'].value else "0"
-        f_openmp = "1" if spec.variants['openmp'].value else "0"
-
-        comp_cxx = comp_for = ''
+    def edit(self, spec, prefix):
+        if '^openblas' in spec:
+            blas = 'openblas'
+        elif '^intel-mkl' in spec:
+            blas = 'mkl'
+        elif '^essl' in spec:
+            blas = 'essl'
+        else:
+            raise InstallError('Supports only BLAS provider '
+                               'openblas, intel-mkl, or essl')
+        config = [
+            'SHELL=bash',
+            'prefix=' + prefix,
+            'mpi=' + ("1" if '+mpi' in spec else "0"),
+            'cuda=' + ("1" if '+cuda' in spec else "0"),
+            'openmp=' + ("1" if '+openmp' in spec else "0"),
+            'blas=' + blas
+        ]
         if '+mpi' in spec:
-            comp_cxx = 'mpicxx'
-            comp_for = 'mpif90'
-
-        make('mpi=' + f_mpi, 'mkl=1', 'cuda=' + f_cuda, 'openmp=' + f_openmp,
-             'CXX=' + comp_cxx, 'FC=' + comp_for)
-        install_tree('lib', prefix.lib)
-        install_tree('test', prefix.test)
-        mkdirp(prefix.include)
-        install('include/slate/slate.hh', prefix.include)
-        install('lapack_api/lapack_slate.hh',
-                prefix.include + "/slate_lapack_api.hh")
-        install('scalapack_api/scalapack_slate.hh',
-                prefix.include + "/slate_scalapack_api.hh")
+            config.extend(['CXX=mpicxx', 'FC=mpif90'])
```

Comment:
I would use `spec['mpi'].mpicxx` and `spec['mpi'].mpifc` here since the exact executable name may vary.